### PR TITLE
Make le16 and le32 truncate on x64

### PR DESCRIPTION
**Pull request by Alan-Jowett**
This pull request primarily focuses on updating the subproject commit in `external/bpf_conformance` and adding new instructions to the `translate` function in both the `vm/ubpf_jit_arm64.c` and `vm/ubpf_jit_x86_64.c` files. The new instructions are designed to handle specific immediate values in the `EBPF_OP_LE` and `EBPF_OP_BE` cases.

Subproject update:

* [`external/bpf_conformance`](diffhunk://#diff-4c83d0aeb8e3e2e0d8a21794f8e46b59c88fa31939be349410b94f4979813a74L1-R1): Updated the subproject commit from `ce9571054cb6e60ed88bddcfc8cbff19dc31dab6` to `032f73030556d08e98669e1eb9d86cce3dddf464`.

Addition of new instructions:

* [`vm/ubpf_jit_arm64.c`](diffhunk://#diff-854aae9d8e9fac050d2acfa7745f604fa16e119484bdedd095fe441a0a7220f4R1070-R1073): In the `translate` function, added a new condition to handle the case when `inst.imm` equals 32 in the `EBPF_OP_BE` case.
* [`vm/ubpf_jit_x86_64.c`](diffhunk://#diff-faaefdbfc142f09bf9f246a6c93375c692c02451e4f2a58073daa4c350d2c8aeL434-R441): In the `translate` function, added new conditions to handle the cases when `inst.imm` equals 16 or 32 in the `EBPF_OP_LE` case.

Resolves: #487

### CODE REVIEW

File: vm/ubpf_jit_x86_64.c [hunk 1]
```diff
@@ -431,6 +431,14 @@ translate(struct ubpf_vm* vm, struct jit_state* state, char** errmsg)
             break;
 
         case EBPF_OP_LE:
+            if (inst.imm == 16) {
+                /* Truncate to 16 bits */
```
Comment (hawkinsw) [inline vm/ubpf_jit_x86_64.c:435]: Would you be willing to append to this comment (and the one below) a short note saying why we do not actually need to do any swap here?

**Review by hawkinsw — COMMENTED** [review]
I was also looking at

https://github.com/iovisor/ubpf/blob/main/vm/ubpf_jit_x86_64.c#L447

and wondering if we need a masking operation there in the 32-bit case?

File: vm/ubpf_jit_x86_64.c (hunk 1 above)
Comment (Alan-Jowett) [reply vm/ubpf_jit_x86_64.c:436]: Done,

**Comment by Alan-Jowett** [comment]
> I was also looking at
> 
> https://github.com/iovisor/ubpf/blob/main/vm/ubpf_jit_x86_64.c#L447
> 
> and wondering if we need a masking operation there in the 32-bit case?

There are now test cases for both be and le where the upper bits of the register are being set and the check verifies they are being truncated.

**Pull request merged**